internal/api/v1/application/part.go

```diff
@@ -111,12 +125,153 @@ func fetchAppChart(c *gin.Context, ctx context.Context, logger logr.Logger, clus
 		"origin", c.Request.URL.String(),
 		"returning", fmt.Sprintf("%d bytes %s as is", contentLength, contentType),
 	)
-	c.DataFromReader(http.StatusOK, contentLength, contentType, reader, nil)
+	c.DataFromReader(http.StatusOK, contentLength, contentType, reader, map[string]string{
+		"X-Content-Length": strconv.FormatInt(response.ContentLength, 10),
+	})
 	return nil
 }
 
-func fetchAppImage(c *gin.Context) apierror.APIErrors {
-	return apierror.NewBadRequestError("image part not yet supported")
+func fetchAppImage(c *gin.Context, ctx context.Context, logger logr.Logger, cluster *kubernetes.Cluster, appRef models.AppRef) apierror.APIErrors {
+	logger.Info("fetching app image")
+
+	// Get application
+	theApp, err := application.Lookup(ctx, cluster, appRef.Namespace, appRef.Name)
+	if err != nil {
+		return apierror.InternalError(err)
+	}
+
+	jobName := fmt.Sprintf("image-export-job-%s-%s-%s", appRef.Namespace, appRef.Name, theApp.StageID)
+	imageOutputFilename := fmt.Sprintf("%s-%s-%s.tar", appRef.Namespace, appRef.Name, theApp.StageID)
+
+	logger.Info("got app chart", "chart image", theApp.ImageURL)
+
+	err = runDownloadImageJob(ctx, cluster, jobName, theApp.ImageURL, imageOutputFilename)
+	if err != nil {
+		return apierror.NewInternalError("failed to create job", "error", err.Error())
+	}
+
+	file, err := getFileImageAndJobCleanup(ctx, cluster, jobName, imageOutputFilename)
+	if err != nil {
+		return apierror.NewInternalError("failed waiting for job done", "error", err.Error())
+	}
+
+	defer func() {
+		err := os.Remove("/image-export/" + imageOutputFilename)
```

Comment:
I see no documentation why this had to be a PVC instead of a simple directory in the server pod.

I believe that the reason is the use of a separate kube job to retrieve the image from the registry. With the PVC both server and job can mount the same (virtual permanent) disk and use it to share data between them.

I strongly believe that this should be spelled out as part of some description/internal documentation of the overall design. This would also include spelling out the reasoning behind the use of a kube job and the introduction of the `skopeo` dependency. Including links to pages explaining skopeo.

Reply:
I will do that. The main reason was to avoid to modify the epinio server image (that is a scratch image with just the epinio binary on it), and use an external job to do this one-time execution. The reason of using `skopeo copy` instead of the `docker save` was to avoid creating a custom image with the docker daemon, while the skopeo binary can be used as a standalone command. Yes, as you said the PVC is used as a shared disk for this operation.
I'll write down in a doc these considerations.

Reply:
The epinio server itself could not be extended to read the image itself ?